include/mbedtls/mbedtls_config.h

```diff
@@ -3634,8 +3634,23 @@
 
 /* Platform options */
 //#define MBEDTLS_PLATFORM_STD_MEM_HDR   <stdlib.h> /**< Header to include if MBEDTLS_PLATFORM_NO_STD_FUNCTIONS is defined. Don't define if no header is needed. */
-//#define MBEDTLS_PLATFORM_STD_CALLOC        calloc /**< Default allocator to use, can be undefined */
-//#define MBEDTLS_PLATFORM_STD_FREE            free /**< Default free to use, can be undefined */
+/** \def MBEDTLS_PLATFORM_STD_CALLOC
+ *
+ * Default allocator to use, can be undefined.
+ * It should initialize the allocated buffer memory to zeroes.
+ * The size of the buffer is the product of the two parameters.
+ * The behavior is undefined if the product of the two parameters overflows size_t.
```

Comment:
(this of course means we're promising that we will never call the function under such circumstances - I'd be surprised if there are no places this could happen)

Reply:
What kind of a test would you like to add?
If the platform memory model permits that and the object pointer size will be bigger than `size_t`, then `calloc(2, SIZE_MAX)` might as well return a non-null pointer.

Reply:
Surely `calloc(SIZE_MAX / 2, SIZE_MAX / 2)` is pretty certain to overflow?

Reply:
I think a C implementation where `calloc` can return a memory block that's larger than `size_t` (and so `calloc` cannot be emulated with `malloc`) is technically valid, but perverse, and I wouldn't mind if our tests failed on such an implementation. Implementations where `size_t` is smaller than available memory do exist, do you know if they actually allow such an allocation? (I16LP32 or IL32P64 or I32LP64, but where `size_t` is the smaller size, which doesn't really have a standard abbreviation.)

Reply:
I honestly don't know. I added a test for `SIZE_MAX/2`.